collection/collection_map.go

```diff
@@ -84,114 +128,61 @@ func (c *Map) FindAll() []types.MatchData {
-		if rx.MatchString(k) {
-			keys = append(keys, k)
-		}
-	}
-	return keys
-}
-
-func (c *Map) keys() []string {
-	var keys []string
-	for k := range c.data {
-		keys = append(keys, k)
-	}
-	return keys
-}
-
-// AddCS a value to some key with case sensitive vKey
-func (c *Map) AddCS(key string, vKey string, vVal string) {
-	aVal := types.AnchoredVar{Name: vKey, Value: vVal}
-	c.data[key] = append(c.data[key], aVal)
-}
-
 // Add a value to some key
-func (c *Map) Add(key string, value string) {
-	c.AddCS(key, key, value)
-}
-
-// AddUniqueCS will add a value to a key if it is not already there
-// with case sensitive vKey
-func (c *Map) AddUniqueCS(key string, vKey string, vVal string) {
-	if c.data[key] == nil {
-		c.AddCS(key, vKey, vVal)
-		return
-	}
-
-	for _, a := range c.data[key] {
-		if a.Value == vVal {
-			return
-		}
-	}
-	c.AddCS(key, vKey, vVal)
-}
-
-// AddUnique will add a value to a key if it is not already there
-func (c *Map) AddUnique(key string, value string) {
-	c.AddUniqueCS(key, key, value)
-}
-
-// SetCS will replace the key's value with this slice
-// internally converts [] string to []types.AnchoredVar
-// with case sensitive vKey
-func (c *Map) SetCS(key string, vKey string, values []string) {
-	c.data[key] = make([]types.AnchoredVar, 0, len(values))
-	for _, v := range values {
-		c.data[key] = append(c.data[key], types.AnchoredVar{Name: vKey, Value: v})
-	}
+func (c *defaultMap) Add(key string, value string) {
+	keyL := strings.ToLower(key)
+	aVal := types.AnchoredVar{Name: key, Value: value}
+	c.data[keyL] = append(c.data[keyL], aVal)
 }
 
 // Set will replace the key's value with this slice
 // internally converts [] string to []types.AnchoredVar
-func (c *Map) Set(key string, values []string) {
-	c.SetCS(key, key, values)
-}
-
-// SetIndexCS will place the value under the index
-// If the index is higher than the current size of the CollectionMap
-// it will be appended
-// with case sensitive vKey
-func (c *Map) SetIndexCS(key string, index int, vKey string, value string) {
-	if c.data[key] == nil {
-		c.data[key] = []types.AnchoredVar{{Name: vKey, Value: value}}
-	}
-	vVal := types.AnchoredVar{Name: vKey, Value: value}
-	if len(c.data[key]) <= index {
-		c.data[key] = append(c.data[key], vVal)
-		return
+func (c *defaultMap) Set(key string, values []string) {
+	keyL := strings.ToLower(key)
+	c.data[keyL] = make([]types.AnchoredVar, 0, len(values))
+	for _, v := range values {
+		c.data[keyL] = append(c.data[keyL], types.AnchoredVar{Name: key, Value: v})
 	}
-	c.data[key][index] = vVal
 }
 
 // SetIndex will place the value under the index
 // If the index is higher than the current size of the CollectionMap
 // it will be appended
-func (c *Map) SetIndex(key string, index int, value string) {
-	c.SetIndexCS(key, index, key, value)
+func (c *defaultMap) SetIndex(key string, index int, value string) {
+	keyL := strings.ToLower(key)
+	values := c.data[keyL]
+	av := types.AnchoredVar{Name: key, Value: value}
```

Comment:
Very small nit: Maybe you want to allocate this once you know for sure you are using it?